src/remoteCollection/collect.js

```diff
@@ -205,7 +211,11 @@ function prepareRemoteRequest(generator) {
     })
     .then((token) => {
       if (token) {
+        genGlobal.updateGeneratorGlobal(generator.name, 'OAuthToken', token);
         generator.OAuthToken = token;
```

Comment:
If we're tracking the token elsewhere we shouldn't need to set the OAuthToken property on the generator object at all - we should instead always use the globally tracked token.